src/pkg/api/user.go

```diff
@@ -294,10 +292,51 @@ func (h *UserHandler) Info(c *gin.Context) {
 
 	userInfo, err := h.service.Info(c, userID)
 	if err != nil {
+		if errors.Is(err, errors.ErrNotFound) {
+			c.AbortWithStatus(http.StatusNotFound)
+			return
+		}
+
 		log.Println("service.Info error: ", err)
 		c.AbortWithStatus(http.StatusInternalServerError)
 		return
 	}
 
 	c.JSON(http.StatusOK, userInfo)
 }
+
+// Info by code
+// @Summary  Get doctor info by code
+// @Description
+// @Tags     USER
+// @Accept   json
+// @Produce  json
+// @Param    code           path      string  true  "The pin code of the requested doctor"
+// @Success  200            {object}  model.UserInfo
+// @Failure  404            "User code is not exist"
+// @Failure  422            "The request could not be understood by the server due to incorrect syntax. The client SHOULD NOT repeat the request without modifications."
+// @Failure  500            "Is returned when an unexpected error occurs while processing a request"
+// @Router   /user/code/{code} [get]
+func (h *UserHandler) InfoByCode(c *gin.Context) {
+	codeString := c.Param("code")
+
+	codeInt, _ := strconv.Atoi(codeString)
+	if codeInt < 10000000 || codeInt > 99999999 {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "request code must contain 8 digits"})
+		return
+	}
```

Comment:
how about "00000001" i think sting more preferable with digit check

Reply:
yeah, you are right, it's possible, fixed